### ISSUE TITLE
Skip setting `replicas` for deployment when autoscaler is on

### ISSUE THREAD

**Issue by t83714** [magda-io/magda#3096]
**Skip setting `replicas` for deployment when autoscaler is on

See this ticket: https://github.com/helm/helm/issues/7090

Seems Helm3 will set replicates even when its value is altered by HPA.

We should Skip setting `replicas` for deployment when autoscaler is on.

e.g. we may want to do something like:

```
{{- if not .Values.autoscaling.enabled }}
  replicas: {{ .Values.replicasCount }}
{{- end }}
```

**Comment by jtslear** [comment]
I don't believe this is accurate.  Though I welcome to be proven incorrect.  According to the documentation, this field will default to 1.  Perhaps removing this will result in the desired behavior of specific versions of Kubernetes?  But for at least versions 1.19 and 1.20, implementing this would unnecessarily bring down a large amount of running Pods.

**Comment by jtslear** [comment]
Belay my above comment.  After much further testing today, some details logged here: https://github.com/kubernetes/kubernetes/issues/25238#issuecomment-917230268  This is indeed the right move.  My apologies for the noise.

**Comment by t83714** [comment]
no problem~ Thanks @jtslear for your testing & sharing your result
